Industrialize deployment of new instances (#744)
ambanum/OpenTermsArchive: ops/README.md

ansible-playbook ops/infra.yml
```

- To setup `OpenTermsArchive` app only:

```
ansible-playbook ops/app.yml
```

Setting up the production infrastructure for publishing on the shared versions repository entails decrypting a private key managed with [Ansible Vault](https://docs.ansible.com/ansible/latest/user_guide/vault.html). It is decrypted with a password that we keep safe. You do not need to decrypt this specific private key on your own production server.

> **Note that executing playbook on production inventory will affect all production servers.**

Reviewer: ```suggestion
Note that executing the playbook on the `production` inventory will affect **all** production servers.
```

> If you want to execute a playbook on a specific server you have to add the `--limit` option to your command with the `hostname` as parameter:

Reviewer: ```suggestion
If you want to execute a playbook on a specific server only, add the `--limit` option with the `hostname` as parameter:
```

```

> ansible-playbook --inventory ops/inventories/production.yml ops/site.yml --limit <hostname>

Reviewer: ```suggestion
ansible-playbook --inventory ops/inventories/production.yml ops/site.yml --limit $hostname
```

ansible-playbook ops/app.yml --tags update
```

### Commands examples

> - Deploy OpenTermsArchive application on all servers declared in `ops/inventories/production.yml` inventory file:

Reviewer: ```suggestion
- Deploy the Open Terms Archive application on all servers declared in the `ops/inventories/production.yml` inventory file:
```

  ```
  ansible-playbook --inventory ops/inventories/production.yml ops/app.yml
  ```

- Check deployment without applying changes for `services-all`:
  ```
  ansible-playbook --inventory ops/inventories/production.yml ops/app.yml --limit services-all --check --diff
  ```

> - Deploy OpenTermsArchive application only on `services-all` server and without applying changes in the infrastructure:

Reviewer: ```suggestion
- Deploy OpenTermsArchive application only on the `services-all` server without applying changes to the infrastructure:
```

  `ansible-playbook ops/app.yml -t stop`
- If everything works, destroy that machine and re-run the entire installation on a clean machine to ensure that your changes do not work by coincidence
  `vagrant destroy && vagrant up && ansible-playbook ops/site.yml`
- Re-check that everything is ok
  `vagrant ssh`, `pm2 logs`…
- Then you can now deploy the changes in production
  `ansible-playbook -i ops/inventories/production.yml ops/site.yml`.

## Initialize a new instance

### Provision a server

If you use [OVH Horizon](https://horizon.cloud.ovh.net/project/instances/), click on the `Launch Instance` button. Then fill, at least, the following fields:
  - `Instance name`.
  - `Source`. Suggested: `Debian 11`.
  - `Flavor`. Suggested: `b2-30-flex`.
  - `Key pair`. Suggested: Your own personal SSH key, to allow you to connect to the freshly created server.

### Add host configuration

Add an entry to the production inventory file `ops/inventories/production.yml` for the created host with the server address and proper variables.

### Create repositories

Create the `snapshot` and `version` repositories, with:

- A `master` branch.

> - The `master` branch should be the default branch.

Reviewer: > Isn't the default `main` now?
>
> At least, I suggest to specify that this branch can be configured in the host vars. Right now it feels like this has to be exactly that name.
